ast/src/parsed/display.rs

```diff
@@ -663,6 +664,45 @@ impl<Expr: Display> Display for SelectedExpressions<Expr> {
     }
 }
 
+impl<E: Display> Display for StructDeclaration<E> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        write!(
+            f,
+            "struct {}{} {{\n{}}}",
+            self.name,
+            type_vars_to_string(&self.type_vars),
+            indent(
+                self.fields
+                    .iter()
+                    .map(|named| format!("{}: {},\n", named.name, named.ty))
+                    .format(""),
+                1
+            )
+        )
+    }
+}
+
+impl<E: Display> Display for StructExpression<E> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        write!(
+            f,
+            "{}{}",
+            self.name,
+            if self.fields.is_empty() {
+                "".to_string()
```

Comment:
I think we should still use `{}` in this case, i.e. `StructName{}`,  otherwise it gets too complicated.